lib/streamlit/elements/exception_proto.py

```diff
@@ -176,37 +173,24 @@ def _get_stack_trace(
 
     """
     # Get and extract the traceback for the exception.
-    extracted_traceback = None  # type: Optional[traceback.StackSummary]
-    if exception_traceback is not None:
-        extracted_traceback = traceback.extract_tb(exception_traceback)
-    elif hasattr(exception, "__traceback__"):
-        # This is the Python 3 way to get the traceback.
+    if not exception_traceback:
         extracted_traceback = traceback.extract_tb(exception.__traceback__)
     else:
-        # Hack for Python 2 which will extract the traceback as long as this
-        # method was called on the exception as it was caught, which is
-        # likely what the user would do.
-        _, live_exception, live_traceback = sys.exc_info()
-        if exception == live_exception:
-            extracted_traceback = traceback.extract_tb(live_traceback)
-        else:
-            extracted_traceback = None
+        extracted_traceback = traceback.extract_tb(exception_traceback)
 
     # Format the extracted traceback and add it to the protobuf element.
-    if extracted_traceback is None:
-        stack_trace = [
-            "Cannot extract the stack trace for this exception. "
-            "Try calling exception() within the `catch` block."
+    stack_trace = [
```

Comment:
This initial assignment to `stack_trace` can be removed since it's guaranteed to be overwritten by either the `if` or `else` body below.